Replace reactable with react-table

seaninryan opened the pull request:
I updated `ImageListView` to use [react-table](https://github.com/tannerlinsley/react-table). 

<img src="http://i.imgur.com/6lmAZEs.png">

I pretty much just did the minimum to get it working with sorting on, etc. but just uses the default `react-table` styles etc.

I'm going to try and see what I can do w.r.t. filtering

If you guys like it I can apply it to all of the other pages too

Commit:
Trying out react-table for ImageListView

ehazlett commented:
Nice!  This is looking good!

Commit:
Get image removal working with react-table

seaninryan commented:
Rebased from `origin/swarm-mode` and got Tom's image removal stuff to work with `react-table`

tombee commented:
Was just wondering if there was a way to not show empty rows.. but looking at it, there seems to be an option to do it.. cool :)

```
minRows: 0, // Ensure this many rows are always rendered, regardless of rows on page
```

It's nice that we can set our own defaults too.. according to the docs:

```javascript
import { ReactTableDefaults } from 'react-table'

Object.assign(ReactTableDefaults, {
  defaultPageSize: 10,
  minRows: 3,
  // etc...
})
```

seaninryan commented:
Hmm @tombee just looking at the [`react-table` page](https://github.com/tannerlinsley/react-table)... looks like they're removing support for `minRows`

Commit:
react-table for ContainerListView

tombee commented on ui/package.json:
```diff
@@ -24,6 +24,7 @@
     "react-transform-catch-errors": "^1.0.2",
     "react-transform-hmr": "^1.0.4",
     "reactable": "^0.14.1",
+    "react-table": "^3.1.0",
```
Apparently `5.0.0` is the latest release according to: https://www.npmjs.com/package/react-table, think we should upgrade to that?

seaninryan replied:
Ah. I thought 3.1.0 was the latest based off the RELEASES tab on their GitHub project

Can try it out l8r

Commit:
Apply ReactTable across the board, remove react-table

seaninryan commented:
- applied `react-table`/`ReactTable` to all table elements
- removed references to `reactable`
- updated to version 5.0.0 of `react-table`
- I disabled table filtering on all of these pages for now. Will come back to that problem when I get this PR out of the way

Couple of problems that I think I've introduced :confused: :
- clicking through to the `inspect/` pages fails with an error like:
```
Uncaught TypeError: Cannot read property 'id' of undefined
    at ImageInspectView.componentDidMount (ImageInspectView.js:19)
```
- after adding a `service` end up with a blank `ContentPane`
- wasn't able to test out `NodeInspectView.js`, `RegistryInspectView.js`, ServiceInspectView.js` (wasn't able to navigate to those pages). Presume that these won't work as things stand.

tombee commented on ui/src/components/accounts/AccountListView.js:
```diff
@@ -70,22 +70,22 @@ class AccountListView extends React.Component {
-      </Tr>
-    );
-  };
+//  renderAccount = (account) => {
```
Can probably just remove this now if it's no longer required? :)

tombee commented:
@seaninryan 
I don't think you introduced the inspect page issues, I think that was me when I upgraded react-router (`this.props.params.id` -> `this.props.match.params`).  I believe these issues should be fixed in the PR I have open (e.g. https://github.com/shipyard/shipyard/pull/958/files#diff-c9a1ead50e81e6362c24a83a9296307dR18).

I think this is good to go as-is.

tombee commented:
LGTM
